Sources/GRPC/ServerCallContexts/ServerCallContext.swift

```diff
@@ -44,21 +47,53 @@ open class ServerCallContextBase: ServerCallContext {
   public let logger: Logger
   public var compressionEnabled: Bool = true
 
+  /// - Important: While `UserInfo` has value-semantics, this property retrieves from, and sets a
+  ///   reference wrapped `UserInfo`. The contexts passed to interceptors provide the same
+  ///   reference. As such this may be used as a mechanism to pass information between interceptors
+  ///   and service providers.
+  public var userInfo: UserInfo {
+    get {
+      return self.userInfoRef.value
+    }
+    set {
+      self.userInfoRef.value = newValue
+    }
+  }
+
+  /// A reference to an underlying `UserInfo`. We share this with the interceptors.
+  private let userInfoRef: Ref<UserInfo>
+
   /// Metadata to return at the end of the RPC. If this is required it should be updated before
   /// the `responsePromise` or `statusPromise` is fulfilled.
   public var trailers = HPACKHeaders()
 
-  public init(eventLoop: EventLoop, headers: HPACKHeaders, logger: Logger) {
+  public convenience init(
```

Comment:
Not really, to be honest. These context types are only public because they were created as public. They're not _particularly_ useful by themselves. 

A little more context: a subclass of this will be provided to the user provided method handler (i.e. the implementations of methods on their service), but of course the user isn't responsible for creating that context. Indeed the actual context provided is internal. The only time a user would want to create their own context would be to unit test their service provider in isolation, and we provide testing contexts just for that. I guess to sum up: documentation isn't useful because this API isn't really useful. And we're only keeping the API to not break existing users.